packages/syft/src/syft/core/tensor/autodp/intermediate_gamma.py

```diff
@@ -257,3 +258,42 @@ def __mul__(self, other: Any) -> IntermediateGammaTensor:
             bias_tensor=bias_tensor,
             scalar_manager=self.scalar_manager,
         )
+
+    def __eq__(self, other: SupportedChainType) -> IntermediateGammaTensor:
+        if is_acceptable_simple_type(other):
+            term_data = self.term_tensor == other  # Need to check if shapes are broadcastable!!
+        elif isinstance(other, IntermediateGammaTensor):
+            # TODO: Check what actually needs to be identical to do an equality comparison
```

Comment:
Good question. We really should add `__eq__` to the new serde interface because its crucial that equality is defined on a class so that we can confirm that the serialization works as expected. We probably need to do this through the whole tensor chain since each tensor can have arbitrary attributes with values which need comparison.